### ISSUE TITLE
Game Crashes When Using the Supplementaries Cannon on a Ship from the Eureka Ships Mod

### ISSUE THREAD

**Issue by ghost** [ValkyrienSkies/Eureka#425]
### Mod Name

Supplementaries

### This issue occurs when only Valkyrien Skies, addons, and the mod I have specified are installed and no other mods

- [X] I have tested this issue and it occurs with only Valkyrien Skies, addons, and the mod I have specified

### Minecraft Version

1.2

### Mod Loader

Fabric

### Eureka version

[Fabric 1.20.1] v1.5.1-beta.3

### Issue description

> **_Modpack, not just the mods that have the conflict:_**
When I'm on an Assemble ship from the "_Eureka Ship/Valkyrien Skies_" mod and try to use the cannon sights from the Supplementaries mod, either I can't use it and get ejected from the cannon, or I manage to use it but the game crashes instantly. Is there any way to resolve this issue and make the two mods work together in this regard?
***Important note:** This bug only occurs when I try to enter the cannon's aiming mode (_"shift + right-click"_). If I just right-click, it opens the normal interface, and I can even shoot and move the cannon by changing the numbers in the interface. However, if I try to enter the aiming mode, the bug happens.

> **_Only the mods that have the conflict._**
However, I created a new instance with only the mods involved in the reported incompatibility to test it, and the game didn’t crash. Instead, this happened: https://youtu.be/cEIjoclFr8U

In summary, when I used other mods within my modpack, the game crashed (_possibly due to it being too heavy_), but when I only use the mods that cause the conflict, the game doesn't crash, but I still can't access the cannon aim (_as shown in the video_), which is what's desired.

### Issue reproduction

I would like to open the cannon interface while the ship is assembled.

### Logs

Crash log:  https://mclo.gs/4hMW4C2

**Comment by ghost** [comment]
![Desktop Screenshot 2024 11 20 - 20 27 35 46](https://github.com/user-attachments/assets/00094dd8-5919-4364-8502-3dafcfab3db7)
**_In this screenshot, these are the mods present in the instance I mentioned, which I created with only the mods that cause the conflict._**

**Comment by Comm-GT** [comment]
bump because I'm having this same issue. also the cannons are strangely not able to destroy ship's blocks, but it works just fine on other blocks.

**Comment by tnnf** [comment]
In my experience this happens with every block that changes your pov. Another example would be securitycraft cameras. You just get ejected instantly.
